postgres/datadog_checks/postgres/statements.py

```diff
@@ -37,88 +37,55 @@
-PG_STAT_STATEMENTS_METRIC_COLUMNS = {
-    'calls': 'postgresql.queries.count',
-    'total_time': 'postgresql.queries.time',
-    'rows': 'postgresql.queries.rows',
-    'shared_blks_hit': 'postgresql.queries.shared_blks_hit',
-    'shared_blks_read': 'postgresql.queries.shared_blks_read',
-    'shared_blks_dirtied': 'postgresql.queries.shared_blks_dirtied',
-    'shared_blks_written': 'postgresql.queries.shared_blks_written',
-    'local_blks_hit': 'postgresql.queries.local_blks_hit',
-    'local_blks_read': 'postgresql.queries.local_blks_read',
-    'local_blks_dirtied': 'postgresql.queries.local_blks_dirtied',
-    'local_blks_written': 'postgresql.queries.local_blks_written',
-    'temp_blks_read': 'postgresql.queries.temp_blks_read',
-    'temp_blks_written': 'postgresql.queries.temp_blks_written',
-}
-
-# Columns to apply as tags
-PG_STAT_STATEMENTS_TAG_COLUMNS = {
-    'datname': 'db',
-    'rolname': 'user',
-    'query': 'query',
-}
-
-# These limits define the top K and bottom K unique query rows for each metric. For each check run the
-# max metrics sent will be sum of all numbers below (in practice, much less due to overlap in rows).
-DEFAULT_STATEMENT_METRICS_LIMITS = {
-    'calls': (400, 0),
-    'total_time': (400, 0),
-    'rows': (400, 0),
-    'shared_blks_hit': (50, 0),
-    'shared_blks_read': (50, 0),
-    'shared_blks_dirtied': (50, 0),
-    'shared_blks_written': (50, 0),
-    'local_blks_hit': (50, 0),
-    'local_blks_read': (50, 0),
-    'local_blks_dirtied': (50, 0),
-    'local_blks_written': (50, 0),
-    'temp_blks_read': (50, 0),
-    'temp_blks_written': (50, 0),
-    # Synthetic column limits
-    'avg_time': (400, 0),
-    'shared_blks_ratio': (0, 50),
-}
-
-
-def generate_synthetic_rows(rows):
-    """
-    Given a list of rows, generate a new list of rows with "synthetic" column values derived from
-    the existing row values.
-    """
-    synthetic_rows = []
-    for row in rows:
-        new = copy.copy(row)
-        new['avg_time'] = float(new['total_time']) / new['calls'] if new['calls'] > 0 else 0
-        new['shared_blks_ratio'] = (
-            float(new['shared_blks_hit']) / (new['shared_blks_hit'] + new['shared_blks_read'])
-            if new['shared_blks_hit'] + new['shared_blks_read'] > 0
-            else 0
-        )
-
-        synthetic_rows.append(new)
+PG_STAT_STATEMENTS_OPTIONAL_COLUMNS = frozenset({'queryid'})
 
-    return synthetic_rows
+PG_STAT_ALL_DESIRED_COLUMNS = (
+    PG_STAT_STATEMENTS_METRICS_COLUMNS | PG_STAT_STATEMENTS_TAG_COLUMNS | PG_STAT_STATEMENTS_OPTIONAL_COLUMNS
+)
 
 
 class PostgresStatementMetrics(object):
     """Collects telemetry for SQL statements"""
 
-    def __init__(self, config):
-        self.config = config
-        self.log = get_check_logger()
+    def __init__(self, check, config):
+        self._check = check
+        self._config = config
+        self._db_hostname = resolve_db_host(self._config.host)
```

Comment:
Can we avoid making network calls on init?

Reply:
Yes good point, updated to load this lazily